External service returns 429 HTTP error when remote context is resolved (https://www.w3.org/2018/credentials/v1)

### Is this a regression?

No

### Description

Looks like the external resource service returns 429 HTTP error code (too many requests) during the VC issuance flow when fetching the revocation list LD context.

It blocks the issuance flow which is a critical for the basic functionality of the cloud-agent

### Please provide the exception or error you saw

```Markdown
Log of the cloud-agent:
`2025-04-02_05:57:46.295 ERROR o.h.i.a.s.j.S.syncRevocationStatuses@L171:[zio-default-blocking-170] {} - zio-fiber-1103705882 Error processing status list record: c6109269-a92b-4d3b-b6ac-65edc7dd9fd1   
zio.FiberFailure: There wa a problem encountered loading a remote context [https://www.w3.org/2018/credentials/v1]
        at com.apicatalog.jsonld.context.ActiveContextBuilder.fetch(ActiveContextBuilder.java:533)
        at com.apicatalog.jsonld.context.ActiveContextBuilder.create(ActiveContextBuilder.java:172)
        at com.apicatalog.jsonld.expansion.ObjectExpansion.initLocalContext(ObjectExpansion.java:197)
        at com.apicatalog.jsonld.expansion.ObjectExpansion.expand(ObjectExpansion.java:101)
        at com.apicatalog.jsonld.expansion.Expansion.compute(Expansion.java:119)
        at com.apicatalog.jsonld.processor.ExpansionProcessor.expand(ExpansionProcessor.java:124)
        at com.apicatalog.jsonld.processor.ToRdfProcessor.toRdf(ToRdfProcessor.java:69)
        at com.apicatalog.jsonld.api.ToRdfApi.get(ToRdfApi.java:166)
        at org.hyperledger.identus.shared.json.Json$.canonicalizeJsonLDoRdf$$anonfun$1(Json.scala:39)
        at scala.util.Try$.apply(Try.scala:217)
        at org.hyperledger.identus.shared.json.Json$.canonicalizeJsonLDoRdf(Json.scala:46)
        at org.hyperledger.identus.pollux.vc.jwt.EcdsaSecp256k1Signature2019ProofGenerator$.generateProof$$anonfun$1(Proof.scala:59)
        at zio.ZIO$.fromEither$$anonfun$1(ZIO.scala:3680)
Caused by: com.apicatalog.jsonld.JsonLdError: There wa a problem encountered loading a remote context [https://www.w3.org/2018/credentials/v1]
        at com.apicatalog.jsonld.context.ActiveContextBuilder.fetch(ActiveContextBuilder.java:533)
        at com.apicatalog.jsonld.context.ActiveContextBuilder.create(ActiveContextBuilder.java:172)
        at com.apicatalog.jsonld.expansion.ObjectExpansion.initLocalContext(ObjectExpansion.java:197)
        at com.apicatalog.jsonld.expansion.ObjectExpansion.expand(ObjectExpansion.java:101)
        at com.apicatalog.jsonld.expansion.Expansion.compute(Expansion.java:119)
        at com.apicatalog.jsonld.processor.ExpansionProcessor.expand(ExpansionProcessor.java:124)
        at com.apicatalog.jsonld.processor.ToRdfProcessor.toRdf(ToRdfProcessor.java:69)
        at com.apicatalog.jsonld.api.ToRdfApi.get(ToRdfApi.java:166)
        at org.hyperledger.identus.shared.json.Json$.canonicalizeJsonLDoRdf$$anonfun$1(Json.scala:39)
        at scala.util.Try$.apply(Try.scala:217)
        at org.hyperledger.identus.shared.json.Json$.canonicalizeJsonLDoRdf(Json.scala:46)
        at org.hyperledger.identus.pollux.vc.jwt.EcdsaSecp256k1Signature2019ProofGenerator$.generateProof$$anonfun$1(Proof.scala:59)
        at zio.ZIO$.fromEither$$anonfun$1(ZIO.scala:3680)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:999)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
        at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
        at zio.internal.FiberRuntime.evaluateEffect(FiberRuntime.scala:413)
        at zio.internal.FiberRuntime.evaluateMessageWhileSuspended(FiberRuntime.scala:489)
        at zio.internal.FiberRuntime.drainQueueOnCurrentThread(FiberRuntime.scala:250)
        at zio.internal.FiberRuntime.run(FiberRuntime.scala:138)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.apicatalog.jsonld.JsonLdError: Unexpected response code [429]
        at com.apicatalog.jsonld.loader.DefaultHttpLoader.loadDocument(DefaultHttpLoader.java:95)
        at com.apicatalog.jsonld.loader.HttpLoader.loadDocument(HttpLoader.java:6)
        at com.apicatalog.jsonld.loader.SchemeRouter.loadDocument(SchemeRouter.java:62)
        at com.apicatalog.jsonld.context.ActiveContextBuilder.fetch(ActiveContextBuilder.java:529)
        ... 26 common frames omitted`
```

### Please provide the environment you discovered this bug in

```Markdown
Socious reported the issue in the Discord thread
https://discord.com/channels/1146426895114702858/1156506653567496232/1356871112532951225
```

### Anything else?

_No response_

--- a/shared/json/src/main/scala/org/hyperledger/identus/shared/json/Json.scala
+++ b/shared/json/src/main/scala/org/hyperledger/identus/shared/json/Json.scala
@@ -2,6 +2,7 @@ package org.hyperledger.identus.shared.json
 
 import com.apicatalog.jsonld.document.JsonDocument
 import com.apicatalog.jsonld.http.media.MediaType
+import com.apicatalog.jsonld.loader.{DocumentLoader, HttpLoader, LRUDocumentCache}
 import com.apicatalog.jsonld.JsonLd
 import com.apicatalog.rdf.Rdf
 import io.setl.rdf.normalization.RdfNormalize
@@ -36,7 +37,7 @@ object Json {
     Try {
       val inputStream = new ByteArrayInputStream(jsonLdStr.getBytes)
       val document = JsonDocument.of(inputStream)
-      val rdfDataset = JsonLd.toRdf(document).get
+      val rdfDataset = JsonLd.toRdf(document).loader(lruDocumentLoader).get
       val normalized = RdfNormalize.normalize(rdfDataset)
       val writer = new StringWriter
       val rdfWriter = Rdf.createWriter(MediaType.N_QUADS, writer)
@@ -45,4 +46,6 @@ object Json {
       bytes
     }.toEither
   }
+
+  lazy val lruDocumentLoader: DocumentLoader = new LRUDocumentCache(HttpLoader.defaultInstance(), 1024)
 }

--- a/shared/json/src/test/scala/org/hyperledger/identus/shared/json/JsonLdLoadSpec.scala
+++ b/shared/json/src/test/scala/org/hyperledger/identus/shared/json/JsonLdLoadSpec.scala
@@ -1,0 +1,54 @@
+package org.hyperledger.identus.shared.json
+
+import zio.test.*
+import zio.test.Assertion.*
+import zio.ZIO
+import com.apicatalog.jsonld.JsonLd
+import com.apicatalog.jsonld.document.JsonDocument
+import org.hyperledger.identus.shared.json.Json.*
+
+import java.io.ByteArrayInputStream
+
+object JsonLdLoadSpec extends ZIOSpecDefault {
+  
+  private val jsonLdString = """{
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1",
+      {
+        "id": "@id",
+        "type": "@type",
+        "name": "http://schema.org/name"
+      }
+    ],
+    "id": "https://example.com/credentials/123",
+    "type": ["VerifiableCredential", "UniversityDegreeCredential"],
+    "issuer": {
+      "id": "https://example.edu/issuers/565049",
+      "name": "Example University"
+    },
+    "credentialSubject": {
+      "id": "https://example.edu/students/alice",
+      "degree": {
+        "type": ["BachelorDegree", "UniversityDegree"],
+        "name": "Bachelor of Science and Arts"
+      }
+    }
+  }"""
+  
+  val jsonLDDocument = JsonDocument.of(new ByteArrayInputStream(jsonLdString.getBytes))
+
+  def loadJsonLd: ZIO[Any, Throwable, Unit] = {
+    ZIO.attempt {
+      JsonLd.toRdf(jsonLDDocument)
+        .loader(Json.lruDocumentLoader) // It fails with an error without the lru loader
+        .get
+    }.unit
+  }
+
+  override def spec = suite("JsonLdLoadSpec")(
+    test("Execute JsonLd.toRdf 1000 times") {
+      val loadJsonLdRepeatedly = ZIO.foreach(1 to 1000)(_ => loadJsonLd)
+      assertZIO(loadJsonLdRepeatedly.exit)(succeeds(anything))
+    }
+  )
+}

--- a/shared/json/src/test/scala/org/hyperledger/identus/shared/json/JsonLdLoadSpec.scala
+++ b/shared/json/src/test/scala/org/hyperledger/identus/shared/json/JsonLdLoadSpec.scala
@@ -1,16 +1,16 @@
 package org.hyperledger.identus.shared.json
 
+import com.apicatalog.jsonld.document.JsonDocument
+import com.apicatalog.jsonld.JsonLd
+import org.hyperledger.identus.shared.json.Json.*
 import zio.test.*
 import zio.test.Assertion.*
 import zio.ZIO
-import com.apicatalog.jsonld.JsonLd
-import com.apicatalog.jsonld.document.JsonDocument
-import org.hyperledger.identus.shared.json.Json.*
 
 import java.io.ByteArrayInputStream
 
 object JsonLdLoadSpec extends ZIOSpecDefault {
-  
+
   private val jsonLdString = """{
     "@context": [
       "https://www.w3.org/2018/credentials/v1",
@@ -34,12 +34,13 @@ object JsonLdLoadSpec extends ZIOSpecDefault {
       }
     }
   }"""
-  
+
   val jsonLDDocument = JsonDocument.of(new ByteArrayInputStream(jsonLdString.getBytes))
 
   def loadJsonLd: ZIO[Any, Throwable, Unit] = {
     ZIO.attempt {
-      JsonLd.toRdf(jsonLDDocument)
+      JsonLd
+        .toRdf(jsonLDDocument)
         .loader(Json.lruDocumentLoader) // It fails with an error without the lru loader
         .get
     }.unit

